File: php/moderation_long_video_aksk_demo.php
<?php
/**
 * 长视频审核服务ak,sk 方式请求的使用示例
 */
require "./moderation_sdk/moderation_long_video.php";
require "./moderation_sdk/utils.php";

$app_key = "********";
$app_secret = "*********";
// 每个账户的对应region下projectId唯一确定的
$projectId = "*********";

/**
 * 使用url配置方式视频审核
 */
$inputType = "url";
$url = "https://moderation-sdk-video.obs.cn-north-4.myhuaweicloud.com/input/demo.mp4";
$inputParam = get_moderation_input($inputType, $inputBucket=null, $inputPath=null, $url);

// obs输出文件所在桶的名称
$outputBucket = "moderation-sdk-video";
// obs输出文件在桶中的路径
$outputPath = "output/";
$outputParam = get_moderation_output($outputBucket, $outputPath);

// 任务名称
$taskName = "task-demo";
// 任务描述
$description = "description";
$requestBody = get_moderation_requestbody($taskName, $description ,$inputParam, $outputParam, "porn,terrorism,politics",
    "ad,politics,porn,abuse,contraband,flood");

$result = moderation_long_video_aksk($app_key, $app_secret, $projectId, $requestBody);
echo json_encode($result);
echo "\n";

/**
 * 使用obs配置方式视频审核
 */
$inputType = "obs";
$inputBucket = "moderation-sdk-video";
$inputPath = "input/demo.mp4";
$inputParam = get_moderation_input($inputType, $inputBucket, $inputPath, $url=null);

$requestBody = get_moderation_requestbody($taskName, $description ,$inputParam, $outputParam,
    "porn,terrorism,politics", "ad,politics,porn,abuse,contraband,flood");

$result = moderation_long_video_aksk($app_key, $app_secret, $projectId, $requestBody);
echo json_encode($result);

File: python/moderation_long_video_aksk_demo.py
# -*- coding:utf-8 -*-
from moderation_sdk.moderation_long_video import moderation_long_video_aksk,get_moderation_input,get_moderation_output,get_moderation_requestbody
from moderation_sdk.utils import init_global_env
if __name__ == '__main__':
    init_global_env("cn-north-4")

    #
    # access long video moderation,post data by ak,sk
    #
    app_key = '*************'
    app_secret = '*************'

    project_id = '*************'

    input_type = "url"
    url = "https://moderation-sdk-video.obs.cn-north-4.myhuaweicloud.com/input/demo.mp4"
    input_param = get_moderation_input(input_type, None, None, url)

    output_bucket = "moderation-sdk-video"
    output_path = "output/"
    output_param = get_moderation_output(output_bucket, output_path)

    task_name = "task-demo"
    description = "description"
    request_body = get_moderation_requestbody(task_name, description, input_param, output_param,
                                              "porn,terrorism,politics",
                                              "ad,politics,porn,abuse,contraband,flood")
    # call interface use the url
    result = moderation_long_video_aksk(app_key, app_secret, project_id, request_body)
    print(result)

    # call interface use the obs
    input_type = "obs"
    input_bucket = "moderation-sdk-video"
    input_path = "input/demo.mp4"
    input_param = get_moderation_input(input_type, input_bucket, input_path, None)

    request_body = get_moderation_requestbody(task_name, description, input_param, output_param,
                                             "porn,terrorism,politics",
                                             "ad,politics,porn,abuse,contraband,flood")

    result = moderation_long_video_aksk(app_key, app_secret, project_id, request_body)
    print(result)

